tests/test_osm.py

```diff
@@ -1,74 +1,51 @@
-from unittest import TestCase, skip
+from unittest import TestCase
 
-from mappymatch.constructs.coordinate import Coordinate
-from mappymatch.constructs.geofence import Geofence
-from mappymatch.maps.nx.readers.osm_readers import NetworkType, read_osm_nxmap
-from mappymatch.utils.crs import LATLON_CRS, XY_CRS
+import networkx as nx
+
+from mappymatch.maps.nx.readers.osm_readers import (
+    NetworkType,
+    parse_osmnx_graph,
+)
 from tests import get_test_dir
 
 
 class TestOSMap(TestCase):
     def test_osm_networkx_graph_drive(self):
-        gfile = get_test_dir() / "test_assets" / "downtown_denver.geojson"
+        gfile = get_test_dir() / "test_assets" / "osmnx_drive_graph.pickle"
 
-        geofence = Geofence.from_geojson(gfile)
+        osmnx_graph = nx.read_gpickle(gfile)
 
-        osm_map = read_osm_nxmap(geofence)
+        cleaned_graph = parse_osmnx_graph(osmnx_graph, NetworkType.drive)
 
         self.assertEqual(
-            osm_map.g.graph["network_type"], NetworkType.drive.value
+            cleaned_graph.graph["network_type"], NetworkType.drive.value
+        )
+
+        self.assertTrue(
+            nx.is_strongly_connected(cleaned_graph),
+            "Graph is not strongly connected",
         )
 
-        # Make sure we can find 31st St near Mestizo-Curtis Park
-        mestizo_curtis_park = Coordinate.from_lat_lon(39.761613, -104.977189)
-        st31 = osm_map.nearest_road(mestizo_curtis_park.to_crs(XY_CRS))
-        # DEBUG: plot the street it choose using geojson.io
-        # import geopandas
-        # print(geopandas.GeoSeries(st31.geom).set_crs(XY_CRS).to_crs(LATLON_CRS).to_json())
-        possible_ids = ["1321042414-3323569423", "3323569423-1321042414"]
-        self.assertTrue(st31.road_id in possible_ids)
+        has_geom = all(
+            [
+                d.get("geometry") is not None
+                for _, _, d in cleaned_graph.edges(data=True)
+            ]
+        )
+        self.assertTrue(has_geom, "All edges should have geometry")
 
         # Make sure the graph contains this road segment
-        e23_ave_node1 = osm_map.g.nodes[1321042414]
-        e23_ave_node2 = osm_map.g.nodes[3323569423]
+        # TODO: this might be a bit unstable since OSM node ids might change
+        e23_ave_node1 = cleaned_graph.nodes[1321042414]
+        e23_ave_node2 = cleaned_graph.nodes[3323569423]
         self.assertAlmostEqual(e23_ave_node1["lat"], 39.761877, 5)
         self.assertAlmostEqual(e23_ave_node1["lon"], -104.9775733, 5)
         self.assertAlmostEqual(e23_ave_node2["lat"], 39.7602886, 5)
         self.assertAlmostEqual(e23_ave_node2["lon"], -104.9779113, 5)
         self.assertAlmostEqual(
-            osm_map.g.get_edge_data(1321042414, 3323569423, 0)["kilometers"],
+            cleaned_graph.get_edge_data(1321042414, 3323569423, 0)[
+                "kilometers"
+            ],
             0.246787,
             5,
         )
-
-    def test_osm_networkx_graph_walk(self):
```

Comment:
Are we getting rid of this test here?

Reply:
My thinking here is that it was redundant. The primary difference between these two tests is that they download a different network and we're not explicitly testing that part of the code since it lies in the purview of the osmnx package. Once we get the network into our code, we treat it the same regardless of the network type.